src/api/app/controllers/trigger_controller.rb

```diff
@@ -5,36 +5,42 @@ class TriggerController < ApplicationController
   skip_before_action :extract_user
   # Authentication happens with tokens, so no login is required
   skip_before_action :require_login
+  # GitLab sends data as parameters which are not strings, like the project hash.
+  skip_before_action :validate_params
   after_action :verify_authorized
 
-  before_action :validate_gitlab_event
+  before_action :validate_gitlab_event, if: :gitlab_webhook?
   before_action :set_token
   before_action :set_package
 
   include Trigger::Errors
 
+  # TODO: split into different controllers, there is some behaviour that only applies to one specific kind of token.
   def create
     authorize @token
     @token.user.run_as do
-      @token.call(params.slice(:repository, :arch).permit!)
+      # TODO: This is what we should have in place for the Token::Workflow
+      @token.call(params.slice(:repository, :arch).permit!.merge(request: request))
```

Comment:
This change was introduced since I expected that we would need the request object in the `Token::Workflow` model. I prefer this solution over having another controller only for `Token::Workflow`.

Reply:
We opted for a new controller because it was doing too much: half of the controller was dealing with workflow tokens only.

Reply:
Yes, but my point is to pass the request object to the `Token::Workflow` model in the `call` method and everything unique to that model will be handled in the model itself.

Reply:
There are two different things:

1) What arguments should we pass to the `Token::Workflow`? 
Even if we don't split the code into two controllers, we didn't like the idea of passing the whole request to the model. We prefer to pass what the model needs instead. If we don't split the controller, it will be something like:
```@token.call(params.slice(:repository, :arch).permit!.merge({ scm: scm, event: event, payload: payload }))```

2) Having one or two controllers?
We need to do some checks that correspond only to workflow tokens. That's why we considered that checking different conditions was no so cool and having its own controller is :)

And after our decision in 1) we also need to get `scm`, `event` and `payload`, that are only necessary for workflow tokens.

PD: the `.merge(request: request))` was added during the rebase probably, we should remove it if we agree to go on with our approach.

Reply:
I already removed `.merge(request: request))` part from the `TriggerController`. We pass `scm`, `event` and `payload` from `TriggerWorkflowController` instead.

Reply:
@dmarcoux the only thing that won't work is to have workflow tokens reaching the same action as the other tokens. Workflow tokens don't need to run the filter `set_package`. So even if you want to have both in one controller, they won't have the same entrypoint. So i guess, its better a new controller with one action `create` than a custom action name on the `TriggerController`